packages/third-parties/src/google/gtm.tsx

```diff
@@ -0,0 +1,59 @@
+'use client'
+// TODO: Evaluate import 'client only'
+import React from 'react'
+import Script from 'next/script'
+
+declare global {
+  interface Window {
+    dataLayer?: Object[]
+    [key: string]: any
+  }
+}
+
+type GTMParams = {
+  gtmId: string
+  dataLayer: string[]
+  dataLayerName: string
+  auth: string
+  preview: string
+}
+
+let currDataLayerName = 'dataLayer'
+
+export function GoogleTagManager(props: GTMParams) {
+  const { gtmId, dataLayerName = 'dataLayer', auth, preview, dataLayer } = props
```

Comment:
Can we rename `gtmId` to just `id`?

Reply:
I originally had it as id but since I made it into a component, I am accepting the GTM params as props

`<GoogleTagManager gtmId="GTM_XYZ" />`

In this case id might clash with HTML id so I changed it to gtmID